Source/Fluids/WarpXFluidContainer.cpp

```diff
@@ -1055,9 +1068,15 @@ void WarpXFluidContainer::GatherAndPush (
                         By_type, Nodal_type, coarsening_ratio, i, j, k, 0);
                     amrex::Real Bz_Nodal = ablastr::coarsen::sample::Interp(Bz_arr,
                         Bz_type, Nodal_type, coarsening_ratio, i, j, k, 0);
+                    // NOLINTEND(misc-const-correctness)
+
+#ifdef AMREX_USE_CUDA
+                    amrex::ignore_unused(Exfield_parser, Eyfield_parser, Ezfield_parser,
+                                         Bxfield_parser, Byfield_parser, Bzfield_parser, gamma_boost);
+#endif
 
-                    if (gamma_boost > 1._rt) { // Lorentz transform fields due to moving frame
-                        if ( ( external_b_fields ) || ( external_e_fields ) ){
+                    if constexpr (boost_control == has_gamma_boost) { // Lorentz transform fields due to moving frame
+                        if constexpr (exte_control == has_exte || extb_control == has_extb) {
```

Comment:
Wow, NVCC is picky again and needs a few temporaries before we go in the `if constexpr`:
```
error: An extended __device__ lambda cannot first-capture variable in constexpr-if context
```
for
- `problo`
- `dx`
- `t`
- `beta_boost`
- ...

Reply:
Oh, right. We can add them to amrex::ignore_unused.